crates/utils/re_video/src/decode/ffmpeg_cli/ffmpeg.rs

```diff
@@ -1057,6 +1070,58 @@ fn write_avc_chunk_to_nalu_stream(
     Ok(())
 }
 
+fn write_hevc_chunk_to_nalu_stream(
+    hvcc: &re_mp4::HevcBox,
+    out: &mut dyn std::io::Write,
+    chunk: &Chunk,
+    state: &mut NaluStreamState,
+) -> Result<(), Error> {
+    let nal_len_size = (hvcc.hvcc.contents.length_size_minus_one as usize & 0x03) + 1;
+    let mut hvcc_ps: Vec<Vec<u8>> = Vec::new();
+
+    for arr in &hvcc.hvcc.arrays {
+        let t = arr.nal_unit_type;
+        if t == 32 || t == 33 || t == 34 {
+            // VPS/SPS/PPS = 32/33/34
+            for nalu in &arr.nalus {
+                hvcc_ps.push(nalu.data.clone());
+            }
+        }
+    }
+
+    if chunk.is_sync && !state.previous_frame_was_idr {
+        for ps in &hvcc_ps {
+            write_bytes(out, ANNEXB_NAL_START_CODE)?;
+            write_bytes(out, ps)?;
+        }
+        state.previous_frame_was_idr = true;
+    } else {
+        state.previous_frame_was_idr = false;
+    }
+
+    let data = &chunk.data;
+    let mut i = 0usize;
+    while i + nal_len_size <= data.len() {
+        let mut len: usize = 0;
+        for &b in &data[i..i + nal_len_size] {
+            len = (len << 8) | (b as usize);
+        }
+        i += nal_len_size;
+
+        if i + len > data.len() {
+            return Err(Error::BadVideoData("Not enough bytes to".to_owned()));
```

Comment:
both this and the original have this issue 😅 
```suggestion
            return Err(Error::BadVideoData("Video sample data ends with incomplete NAL unit.".to_owned()));
```